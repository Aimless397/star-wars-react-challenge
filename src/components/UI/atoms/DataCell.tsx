interface Props {
  text: string;
  value?: string;
}

export const DataCell = ({ text, value }: Props) => {
  return (
    <div className="dataCellContainer">
      <span data-testid="dataCellCategoryText" className="dataCellCategoryText">
        {text}
      </span>
      {value && <span className="dataCellValueText">{value}</span>}
    </div>
  );
};
