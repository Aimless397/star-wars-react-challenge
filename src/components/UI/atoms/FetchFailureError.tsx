export const FetchFailureError = () => {
  return (
    <div
      data-testid="fetchContentErrorContainer"
      className="fetchContentErrorContainer"
    >
      <span>Please reload the page</span>
    </div>
  );
};
