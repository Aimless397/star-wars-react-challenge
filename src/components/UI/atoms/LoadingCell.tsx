export const LoadingCell = () => {
  return (
    <div className="loadingContainer">
      <div className="spinner-border spinner-border-sm" />
      <span data-testid="loadingSpan" className="loadingSpan">
        Loading
      </span>
    </div>
  );
};
