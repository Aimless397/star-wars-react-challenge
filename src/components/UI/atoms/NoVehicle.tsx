export const NoVehicle = () => {
  return (
    <div className="noVehicleContainer">
      <span data-testid="noVehicleSpan" className="noVehicleSpan">
        No vehicles found
      </span>
    </div>
  );
};
