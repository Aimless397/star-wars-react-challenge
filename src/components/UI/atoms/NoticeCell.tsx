export const NoticeCell = () => {
  return (
    <div className="noticeContainer">
      <span data-testid="noticeSpan" className="noticeSpan">
        Failed to Load Data
      </span>
    </div>
  );
};
