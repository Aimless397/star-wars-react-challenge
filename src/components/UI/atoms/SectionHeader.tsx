interface Props {
  text: string;
}

export const SectionHeader = ({ text }: Props) => {
  return (
    <div className="sectionContainer">
      <span data-testid="sectionHeader" className="sectionHeader">
        {text}
      </span>
    </div>
  );
};
