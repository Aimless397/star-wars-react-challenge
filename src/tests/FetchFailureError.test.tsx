import { describe, expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { FetchFailureError } from "../components/UI/atoms/FetchFailureError";

describe("<FetchFailureError />", () => {
  test("should be rendered", () => {
    const { container } = render(<FetchFailureError />);

    expect(container).toMatchSnapshot();
  });

  test("should have internal text", () => {
    render(<FetchFailureError />);

    expect(screen.getByText("Please reload the page"));
  });

  test("should have the correct className style", () => {
    render(<FetchFailureError />);

    const divElement = screen.getByTestId("fetchContentErrorContainer");

    expect(divElement.className).toBe("fetchContentErrorContainer");
  });
});
