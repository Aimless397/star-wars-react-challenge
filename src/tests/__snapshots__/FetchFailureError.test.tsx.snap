// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FetchFailureError /> should be rendered 1`] = `
<div>
  <div
    class="fetchContentErrorContainer"
    data-testid="fetchContentErrorContainer"
  >
    <span>
      Please reload the page
    </span>
  </div>
</div>
`;
